gninasrc/lib/cnn_scorer.cpp

```diff
@@ -295,6 +295,84 @@ void CNNScorer::get_net_output(Dtype& score, Dtype& aff, Dtype& loss) {
 
   loss = lossblob->cpu_data()[0];
 }
+
+// Extract ligand atoms and coordinates
+std::pair<atomv, vecv> CNNScorer::get_ligand(const model& m) const{
```

Comment:
Ligand atoms are between flex and inflex atoms. This utility function extract ligand atoms and coordinates. There might be better ways of doing this.

Reply:
you don't need to create a temporary copy for every `CNNScorer::score` call; if you're going to rely on having a vector of the receptor-only and ligand-only data, make it a class member

Reply:
Yes, I agree. See the comment above for `setReceptor`.

Reply:
Transformed in `CNNScorer::setReceptor`, storing atoms and coordinates only once (coordinates still need to be updated, to be discussed...)